apps/simple-trading-app/src/app/components/simple-market-list/simple-market-list.spec.tsx

```diff
@@ -50,14 +49,13 @@ describe('SimpleMarketList', () => {
         data: { markets: [] },
       },
     };
-    await act(async () => {
-      render(
-        <MockedProvider mocks={[mocks, filterMock]}>
-          <SimpleMarketList />
-        </MockedProvider>
-      );
-      await new Promise((resolve) => setTimeout(resolve, 0));
-    });
+
```

Comment:
please don't change anything in these tests (console-lite). Current version in master is valid. Another ongoing PR has next changes there, so it here can cause big conflicts

Reply:
FYI, these tests were failing when I edited them. since then, they have been updated on master, yes, but they are still full of ambiguous and sometime redundant hacks imo. i was trying to add some improvements to them while merging the current master.
the thing i don't really agree with is that you're basically saying that I should spend extra time removing my improvements, so it doesn't potentially cause someone else to spend time with resolving conflicts some time in the future. not sure if I agree with this approach. PRs should be valid in isolation, it makes life a lot easier.

Reply:
yeah, this is exactly what i'm asking for here: the isolation for the our PR's.
OK, It doesn't matter, i can resolve all conflicts, np.
By the way, all so called by you "hacks" are wholly taken from apollo client documentation site. What is really a hack can sometimes be deceptive. I would be more careful with such statements.

Reply:
i feel like this is heading toward a direction of being a misunderstanding. I am not trying to insult anyone's code here. I just pointed out that the solutions in those tests are not ideal, and I had to spend time trying to figure out why my PR is being affected by them.

if there's a problem with my changes, I would like to understand what is it. it just didn't seem like a well explained reason that this might cause conflicts in some future PRs. 

the thing with setTimeout here is that it pushes a callback to the event loop, but the component being tested does a lot of things and there are a lot of useEffect calls which also add callbacks to the event loop, i don't think the current approach guarantees the execution order at all. this is why i decided to remove the Promise setTimeout, because it felt like a hack. i wouldn't have touched it if the tests passed, tbh. anyway, tests are passing now without my changes (which wasn't the case ), i will go back here and remove this if it's causing you difficulties, it is not worth the argument imo.

Reply:
Idea of Promise with setTimeout comes streightly from documentation https://www.apollographql.com/docs/react/development-testing/testing/ . Tests with this solution went smoothly, that's why they has been commited and merged. Callback with await clause just should "cut off" rendering phase from testing phase. Synchonize these two asychnonous process. I can understand what was the idea of the authors of this solution. And it even worked, however, as it turned out, not always :-(

And about possible conflics in repo, I was thinking more about your situation in case when my current PR would merge first. I made there many changes not only in unit test but also in component, so more problems would be at your side in such a case.